streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -176,6 +176,19 @@ abstract class ZSink[-R, +E, -I, +L, +Z] private (
   ): ZSink[R1, E1, I2, L, Z2] =
     contramapChunksM(f).mapM(g)
 
+  /**
+   * Catches potential error into `Either`
+   */
+  def either: ZSink[R, Nothing, I, L, Either[E, Z]] = ZSink {
```

Comment:
Can this be implemented in terms of ZSink#foldM?

Reply:
I tried `foldM(e => ZSink.succeed(Left(e)), z => ZSink.succeed(Right(z)))`.
Unfortunately, leftovers typed don't match.